test/e2e/clusterctl_upgrade_test.go

```diff
@@ -95,6 +99,14 @@ var _ = Describe("When testing clusterctl upgrades (v0.3=>v1.5=>current)", Flake
 				},
 				{ // Upgrade to latest v1beta1.
 					Contract: clusterv1.GroupVersion.Version,
+					PostUpgrade: func(proxy framework.ClusterProxy, namespace, clusterName string) {
+						framework.ValidateCRDMigration(ctx, proxy, namespace, clusterName,
+							func(crd apiextensionsv1.CustomResourceDefinition) bool {
+								return strings.HasSuffix(crd.Name, ".cluster.x-k8s.io") &&
+									crd.Name != "providers.clusterctl.cluster.x-k8s.io"
```

Comment:
Not sure what we want to do with providers. At the moment it doesn't matter because it's v1alpha3 and we don't touch it at all